asm/kanread.s

```diff
@@ -0,0 +1,217 @@
+#include "ultra64/asm.h"
+
+// assembler directives
+.set noat
+.set noreorder
+
+.section .text
+
+.balign 16
+
+/**
+ * s32 Kanji_OffsetFromShiftJIS(s32 sjis);
```

Comment:
I weigh this up carefully, decided that u32 would make sense, then looked in the libleo headers and discovered it had this prototype.

Reply:
I see, then I guess we should probably keep this until we match code using it in other versions, but I would still support changing them to u32 eventually if they end up working (and maybe even size_t for the return type?)

Reply:
`size_t` is probably the correct type for the return. But it is worth bearing in mind that `n64dd` on N64 versions has another version of this function: https://github.com/Roman971/oot-disassembly/blob/master/system/n64dd.s#L9577-L9797 , apparently straight from `libleo.a`, for which we'll want to use the headers anyway, so it might be better to stay consistent (assuming that all this matches anyway).